components/timer/countdown.tsx

```diff
@@ -0,0 +1,48 @@
+"use client";
+
+import React from "react";
+
+//TODO: add more display formats
+
+interface Props {
+  endTimestamp: bigint;
+  displayFormat?: "dd:hh:mm:ss";
+}
+
+function getTimeLeft(endTimestamp: bigint): bigint {
```

Comment:
The getTimeLeft returns a bigint, but the time calculations are done with regular numbers. This could lead to precision issues. It would be better to do all the calculations with bigints and only convert to a regular number when displaying the countdown.

Reply:
calculations are done in bigint. Normal math operations are okay if both sides of expression are bigints